File: desktop/primitives.py
PRIMITIVES = {
    'SELF':  {'syllable': 'sa', 'key': 'f', 'emoji': '👤', 'meaning': 'I, me, my'},
    'OTHER': {'syllable': 'to', 'key': 'd', 'emoji': '👥', 'meaning': 'you, they, it'},
    'NOT':   {'syllable': 'na', 'key': 'j', 'emoji': '❌', 'meaning': 'no, negation, opposite'},
    'WANT':  {'syllable': 'wu', 'key': 'k', 'emoji': '🤲', 'meaning': 'desire, need, wish'},
    'GOOD':  {'syllable': 'go', 'key': 's', 'emoji': '✅', 'meaning': 'positive, correct, safe'},
    'BAD':   {'syllable': 'bi', 'key': 'l', 'emoji': '⚠️',  'meaning': 'negative, wrong, danger'},
    'NOW':   {'syllable': 'ne', 'key': 'g', 'emoji': '⚡', 'meaning': 'present time, immediate'},
    'HERE':  {'syllable': 'hi', 'key': 'h', 'emoji': '📍', 'meaning': 'this place, location'},
    'ALIVE': {'syllable': 'li', 'key': 'e', 'emoji': '💚', 'meaning': 'living, active, conscious'},
    'MOVE':  {'syllable': 'mo', 'key': 'r', 'emoji': '➡️', 'meaning': 'motion, action, go'},
    'WATER': {'syllable': 'du', 'key': 'i', 'emoji': '💧', 'meaning': 'liquid, fluid'},
    'AIR':   {'syllable': 're', 'key': 'o', 'emoji': '🌬️', 'meaning': 'gas, breath, sky, wind'},
    'HEAT':  {'syllable': 'fu', 'key': 'w', 'emoji': '🔥', 'meaning': 'temperature, energy, fire'},
    'SOLID': {'syllable': 'ko', 'key': 'v', 'emoji': '🪨', 'meaning': 'physical object, earth, hard'},
    'LIGHT': {'syllable': 'lu', 'key': 'n', 'emoji': '☀️', 'meaning': 'brightness, visible, color'},
    'THINK': {'syllable': 'te', 'key': 'u', 'emoji': '💭', 'meaning': 'know, understand, believe'},
}

KEY_MAP      = {v['key']: k      for k, v in PRIMITIVES.items()}
SYLLABLE_MAP = {v['syllable']: k for k, v in PRIMITIVES.items()}

# Common English words that map directly to primitives (for voice input)
WORD_MAP = {
    'i': 'SELF', 'me': 'SELF', 'my': 'SELF',
    'you': 'OTHER', 'they': 'OTHER', 'it': 'OTHER', 'he': 'OTHER', 'she': 'OTHER',
    'no': 'NOT', 'not': 'NOT', 'never': 'NOT',
    'want': 'WANT', 'need': 'WANT', 'wish': 'WANT',
    'good': 'GOOD', 'yes': 'GOOD', 'ok': 'GOOD', 'okay': 'GOOD', 'right': 'GOOD',
    'bad': 'BAD', 'wrong': 'BAD', 'danger': 'BAD',
    'now': 'NOW', 'today': 'NOW', 'soon': 'NOW',
    'here': 'HERE', 'place': 'HERE', 'where': 'HERE',
    'alive': 'ALIVE', 'live': 'ALIVE', 'life': 'ALIVE',
    'go': 'MOVE', 'move': 'MOVE', 'come': 'MOVE',
    'water': 'WATER', 'drink': 'WATER', 'wet': 'WATER',
    'air': 'AIR', 'wind': 'AIR', 'breath': 'AIR', 'sky': 'AIR',
    'heat': 'HEAT', 'hot': 'HEAT', 'fire': 'HEAT', 'warm': 'HEAT',
    'solid': 'SOLID', 'rock': 'SOLID', 'hard': 'SOLID', 'earth': 'SOLID',
    'light': 'LIGHT', 'bright': 'LIGHT', 'dark': 'LIGHT', 'see': 'LIGHT',
    'think': 'THINK', 'know': 'THINK', 'believe': 'THINK', 'understand': 'THINK',
}

# Learner model: tracks which primitives are acquired
DEFAULT_LEARNER = {p: {'acquired': False, 'exposure': 0, 'correct': 0, 'attempts': 0}
                   for p in PRIMITIVES}

File: mobile/primitives.js
export const PRIMITIVES = {
  SELF:  { syllable: 'sa', key: 'F', emoji: '👤', meaning: 'I, me, my',           color: '#4CAF50' },
  OTHER: { syllable: 'to', key: 'D', emoji: '👥', meaning: 'you, they, it',        color: '#2196F3' },
  NOT:   { syllable: 'na', key: 'J', emoji: '❌', meaning: 'no, negation',          color: '#F44336' },
  WANT:  { syllable: 'wu', key: 'K', emoji: '🤲', meaning: 'desire, need',          color: '#FF9800' },
  GOOD:  { syllable: 'go', key: 'S', emoji: '✅', meaning: 'positive, correct',     color: '#8BC34A' },
  BAD:   { syllable: 'bi', key: 'L', emoji: '⚠️', meaning: 'negative, wrong',       color: '#FF5722' },
  NOW:   { syllable: 'ne', key: 'G', emoji: '⚡', meaning: 'present, immediate',    color: '#FFC107' },
  HERE:  { syllable: 'hi', key: 'H', emoji: '📍', meaning: 'this place',            color: '#00BCD4' },
  ALIVE: { syllable: 'li', key: 'E', emoji: '💚', meaning: 'living, conscious',     color: '#4CAF50' },
  MOVE:  { syllable: 'mo', key: 'R', emoji: '➡️', meaning: 'motion, action',        color: '#9C27B0' },
  WATER: { syllable: 'du', key: 'I', emoji: '💧', meaning: 'liquid, fluid',         color: '#03A9F4' },
  AIR:   { syllable: 're', key: 'O', emoji: '🌬️', meaning: 'gas, breath, sky',      color: '#B3E5FC' },
  HEAT:  { syllable: 'fu', key: 'W', emoji: '🔥', meaning: 'temperature, fire',     color: '#FF5722' },
  SOLID: { syllable: 'ko', key: 'V', emoji: '🪨', meaning: 'object, earth, hard',   color: '#795548' },
  LIGHT: { syllable: 'lu', key: 'N', emoji: '☀️', meaning: 'brightness, visible',   color: '#FFEB3B' },
  THINK: { syllable: 'te', key: 'U', emoji: '💭', meaning: 'know, understand',      color: '#E91E63' },
};

export const NAMES = Object.keys(PRIMITIVES);

// Acquisition tiers — learn in this order
export const TIERS = [
  ['SELF', 'OTHER', 'NOT', 'WANT'],
  ['GOOD', 'BAD', 'NOW', 'HERE'],
  ['ALIVE', 'MOVE', 'THINK', 'WATER'],
  ['AIR', 'HEAT', 'SOLID', 'LIGHT'],
];
